src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,215 @@
+        String query = "UPDATE cars "
+                + "SET cars.model = ?,"
+                + "cars.manufacturer_id = ? "
+                + "WHERE cars.id = ? "
+                + "AND cars.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateStatment = connection.prepareStatement(query)) {
+            updateStatment.setString(1, car.getModel());
+            updateStatment.setLong(2, car.getManufacturer().getId());
+            updateStatment.setLong(3, car.getId());
+            updateStatment.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't update car " + car, e);
+        }
+        removeDriversFromCar(car);
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars "
+                + "SET cars.is_deleted = TRUE "
+                + "WHERE cars.id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStatment = connection.prepareStatement(query)) {
+            deleteStatment.setLong(1, id);
+            return (deleteStatment.executeUpdate() >= 1);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT drivers.id, drivers.name, drivers.license_number, "
+                + "cars_drivers.car_id "
+                + "FROM drivers "
+                + "INNER JOIN cars_drivers "
+                + "ON drivers.id = cars_drivers.driver_id "
+                + "WHERE driver_id = ?;";
+        List<Long> carsID = new ArrayList<>();
+        List<Car> carList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllByDriverStatement = connection.prepareStatement(query)) {
+            getAllByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                Long carID = resultSet.getObject("cars_drivers.car_id", Long.class);
+                carsID.add(carID);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get cars by driver id " + driverId, e);
+        }
+        for (Long carID : carsID) {
+            carList.add(get(carID).get());
+        }
+        return carList;
+    }
+
+    private Car getCar(ResultSet resultSet) throws SQLException {
+        Car car = new Car();
+        car.setId(resultSet.getObject("cars.id", Long.class));
+        car.setModel(resultSet.getString("cars.model"));
+        Manufacturer manufacturer = new Manufacturer(
+                resultSet.getObject("manufacturer_id", Long.class),
```

Comment:
The column alias should be 'manufacturers.id' in the resultSet.getObject() method.